nodesharing-lib/src/main/java/com/redhat/jenkins/nodesharing/RestEndpoint.java

```diff
@@ -248,15 +255,27 @@ protected boolean shouldFail(@Nonnull StatusLine sl) {
             @Override
             protected @CheckForNull CrumbResponse consumeEntity(@Nonnull HttpResponse response) throws IOException {
                 if (response.getStatusLine().getStatusCode() == 404) return null;
-                return createEntity(response, CrumbResponse.class);
+                CrumbResponse result = createEntity(response, CrumbResponse.class);
+                if(result != null) {
+                    Header[] cookie_headers = response.getHeaders("Set-Cookie");
+                    for(Header cookie : cookie_headers) {
+                        result.addCookie(cookie.getValue());
```

Comment:
Rather than amending the response, this can be just written into `getCrumbHeader`'s local variable typed as `List`.